lib/money/parser/fuzzy.rb

```diff
@@ -0,0 +1,174 @@
+      # decimals for the fractional part a currency has or the incorrect notion a currency has a defined decimal
+      # separator (this is a property of the locale). While these heuristics can lead to the expected result for some
+      # cases, the other cases can lead to surprising results such as parsed amounts being 1000x larger than intended.
+      # @deprecated Use {LocaleAware.parse} or {Simple.parse} instead.
+      # @param input [String]
+      # @param currency [String, Money::Currency, nil]
+      # @param strict [Boolean]
+      # @return [Money]
+      # @raise [MoneyFormatError]
+      def parse(input, currency = nil, strict: false)
+        currency = Money::Helpers.value_to_currency(currency)
+        amount = extract_amount_from_string(input, currency, strict)
+        Money.new(amount, currency)
+      end
+
+      private
+
+      def extract_amount_from_string(input, currency, strict)
+        unless input.is_a?(String)
+          return input
+        end
+
+        if input.strip.empty?
+          return '0'
+        end
+
+        number = input.scan(NUMERIC_REGEX).flatten.first
+        number = number.to_s.strip
+
+        if number.empty?
+          if Money.config.legacy_deprecations && !strict
+            Money.deprecate("invalid money strings will raise in the next major release \"#{input}\"")
+            return '0'
+          else
+            raise MoneyFormatError, "invalid money string: #{input}"
+          end
+        end
+
+        marks = number.scan(/[#{ESCAPED_MARKS}]/).flatten
+        if marks.empty?
+          return number
+        end
+
+        if marks.size == 1
+          return normalize_number(number, marks, currency)
+        end
+
+        # remove end of string mark
+        number.sub!(/[#{ESCAPED_MARKS}]\z/, '')
+
+        if amount = number[DOT_DECIMAL_REGEX] || number[INDIAN_NUMERIC_REGEX] || number[CHINESE_NUMERIC_REGEX]
+          return amount.tr(ESCAPED_NON_DOT_MARKS, '')
+        end
+
+        if amount = number[COMMA_DECIMAL_REGEX]
+          return amount.tr(ESCAPED_NON_COMMA_MARKS, '').sub(',', '.')
+        end
+
+        if Money.config.legacy_deprecations && !strict
+          Money.deprecate("invalid money strings will raise in the next major release \"#{input}\"")
+        else
+          raise MoneyFormatError, "invalid money string: #{input}"
+        end
+
+        normalize_number(number, marks, currency)
+      end
+
+      def normalize_number(number, marks, currency)
+        digits = number.rpartition(marks.last)
+        digits.first.tr!(ESCAPED_MARKS, '')
+
+        if last_digits_decimals?(digits, marks, currency)
+          "#{digits.first}.#{digits.last}"
+        else
+          "#{digits.first}#{digits.last}"
+        end
+      end
+
+      def last_digits_decimals?(digits, marks, currency)
+        # Thousands marks are always different from decimal marks
```

Comment:
Nitpick:  `Grouping` marks, not `Thousands` marks.  In locales that adopt the "Chinese" 4-digit grouping system, one group is a myriad, not a thousand.

Reply:
TIL!